marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mastering/default/main.sjs

```diff
@@ -1,20 +1,48 @@
 const mastering = require("/com.marklogic.smart-mastering/process-records.xqy");
+const masteringCollections = require("/com.marklogic.smart-mastering/impl/collections.xqy");
 const masteringConsts = require("/com.marklogic.smart-mastering/constants.xqy");
 
 function main(content, options) {
   const filteredContent = [];
+  // set the target entity based off of the step options
+  options.mergeOptions.targetEntity = options.targetEntity;
+  options.matchOptions.targetEntity = options.targetEntity;
+  // provide default empty array values for collections to simplify later logic
+  options.mergeOptions.collections = Object.assign({"content": [], "archived": []},options.mergeOptions.collections);
+  options.matchOptions.collections = Object.assign({"content": []},options.matchOptions.collections);
+  // sanity check the collections set for the match/merge options
+  if (options.matchOptions.collections.content.length) {
```

Comment:
can we get a null check for match/mergeOptions?

Reply:
Added a check for missing match/mergeOptions